### DISCUSSION TITLE
Logic behind "Automatically play first media item"

### DISCUSSION THREAD

**Discussion by ux-git** [sircharlo/meeting-media-manager#2355] [Q&A] [answered]
Hello! I'm trying to set up a countdown video to be automatically played before the meeting. Could you please explain the nuances of how it works? 

It seems simple at first sight, but now I noticed that the video won't be played if you haven't entered the Media Presentation Mode. Also, I'm trying to test it right now, therefore I'm editing the meeting start time, and re-starting the app each time, but so far I've managed to see the countdown timer only a few times, though I tried more than 10 times 🤔

 Is there anything else important that users should know? It would be great to add this to the documentation website as well.

Example:

1. I've added the countdown video to the recurring media.
2. I've set my meeting to be **today** at 15:55 _(in 15 minutes)_. 
3. Music is set to auto-stop _1 minute before_ the meeting. The "Automatically play first media item" is set to _5 minutes before_ the meeting, the video is a 5:00 countdown video.
4. Restarted the app
5. I start the music and open the media presentation mode.
6. I wait for the countdown to appear, but I can't see any consistency in displaying the video. Sometimes it works, but most of the time not. The music countdown is already 0:05, but the countdown hasn't appeared at all.

**Comment by e-Feitosa** [comment]
Brother, i will show you what I did here.
I added a **recurring media** with 2 minutes, and put it in the order 00-00-00.
![image](https://github.com/sircharlo/meeting-media-manager/assets/90984283/782bd8f6-d534-435d-9320-e341f31a1ef4)

In the settings in **MS (Media setup)**, I checked the option to automatically play the first media in the list (see screenshot), i selected the time 2 minutes.
![image](https://github.com/sircharlo/meeting-media-manager/assets/90984283/88c42612-4755-4120-aaa7-433ffc805b46)

Ready.
![image](https://github.com/sircharlo/meeting-media-manager/assets/90984283/de6aeb81-4185-4a12-9074-2b60ba5d2027)

**Observation:**
Make sure the meeting time in **MS (Meeting setup)** is correct, and the general time on your PC.

**Reply by ux-git** [reply]
Thanks. The time on our PC was 30-40 seconds off, so this tip really helped to fix some confusion.

What about the Media Presentation mode? Should it be turned ON in advance? Or is there any setting that will allow to start the video even if I'm not in the presentation mode?

Also, @mtdvlpr – what do you think about having the ability to start background music playback WHILE a media is being displayed? For instance, after the countdown video started, a brother forgot to turn on the music. In this case, it will be useful.

**Reply by sircharlo** [reply]
Perhaps I'm mistaken, but I'm not sure our current direction makes provision for countdown videos before normal congregation meetings, at least in our branch territory.

**Reply by ux-git** [reply]
Thanks for the answer @sircharlo. The question is whether it's forbidden or whether each congregation is allowed to create something that goes along with the current directions? For example, to keep the year text visible before and after meetings. Therefore, I integrated the countdown most naturally, with no distractions from the year text, but providing a quite useful countdown. 

<img width="1352" alt="Screenshot 2024-02-11 at 19 35 05" src="https://github.com/sircharlo/meeting-media-manager/assets/80993061/398f016f-fe05-45cd-95a3-62fe9fdd5f13">

I discussed this with our coordinator, and while there are no instructions that say to use the countdown, at the same time there seems to be no prohibition against doing so. If there's anything else that we need to consider, please feel free to share.

**Reply by sircharlo** [reply]
I think the S-141, pars. 8 and 9, is ultimately what guided us.

**Reply by ux-git** [reply]
Hmm, though it's public talk reminders, and many points are mostly focused on not distracting the attention from the Bible **while** a talk is being given, it's still good info to consider and discuss with brothers. Perhaps, in the future, we will receive clearer directions about the before-the-meeting media, like a countdown (if there is a need, of course). Thank you Olivier!

**Reply by sircharlo** [reply]
Yes, that's correct, especially when combined with _sfl_ 20:23.

**Comment by e-Feitosa** [comment, marked as answer]
> E o modo de Apresentação de Mídia? Deve ser ativado com antecedência? Ou existe alguma configuração que permita iniciar o vídeo mesmo que eu não esteja no modo de apresentação?
> 
The option to play the first media only works when in presentation mode.
